src/memtable.rs

```diff
@@ -1176,9 +1213,64 @@ impl<A: AllocatorTrait> MemTableRecoverContext<A> {
 
     pub fn merge_append_context(&self, append: MemTableRecoverContext<A>) {
         self.memtables
-            .apply_append_writes(append.log_batch.clone().drain());
+            .apply_append_writes(append.tombstone_items.into_iter());
         self.memtables.merge_append_table(append.memtables);
     }
+
+    #[inline]
+    fn is_tombstone(item: &LogItem) -> bool {
+        match &item.content {
+            LogItemContent::Command(Command::Clean)
+            | LogItemContent::Command(Command::Compact { .. }) => true,
+            LogItemContent::Kv(KeyValue { op_type, .. }) if *op_type == OpType::Del => true,
+            _ => false,
+        }
+    }
+
+    fn accept_new_group(&mut self, queue: LogQueue, id: u64, mut new_group: PendingAtomicGroup) {
+        assert_eq!(queue, LogQueue::Rewrite);
+        if let Some(groups) = self.pending_atomic_groups.get_mut(&id) {
+            let group = groups.last_mut().unwrap();
+            match (group.status, new_group.status) {
+                (AtomicGroupStatus::End, AtomicGroupStatus::Begin) => {
+                    groups.push(new_group);
+                }
+                // (begin, begin), (middle, begin)
+                (_, AtomicGroupStatus::Begin) => {
+                    warn!("discard atomic group: {group:?}");
```

Comment:
Why just discard?

Reply:
The left hand side group is incomplete, we will assume it is caused by a power off.